test/testnullpointer.cpp

```diff
@@ -1361,6 +1362,27 @@ class TestNullPointer : public TestFixture {
         ASSERT_EQUALS("[test.cpp:4] -> [test.cpp:3]: (warning, inconclusive) Either the condition 'if(values)' is redundant or there is possible null pointer dereference: values.\n", errout.str());
     }
 
+    void nullpointer31() { // #8482
+        check("typedef struct\n"
```

Comment:
I think this code example is too long. Imho please avoid typedefs here. We simplify typedefs in the Tokenizer so the ValueFlow does not need to handle them.

In the test cases I prefer that such code is replaced:

    if (foo)
    {
    }

with:

    if (foo) {}

that is only 1 line instead of 3.

Reply:
Ah sorry, I was just copying it directly from the ticket. I will simplify it.